### DISCUSSION TITLE
What are some good first issues for potential contributors?

### DISCUSSION THREAD

**Discussion by cliffordfajardo** [jrestall/remix-kit#6] [Q&A]
Hey @jrestall,
I like the work you're doing with remix-kit
Are there currently issues that you might be good for potential new contributors to this project? 
I'm currently onboarding and learning lots of the tool's you're using in this stack

Currently I've been trying out the package and sharing what I've encountered along the journey of using your package to try to smoothen the onboarding experience for others who may try it -- hope your ok with lots of small bug / question reports from my end 😅

**Comment by jrestall** [comment]
Hi @cliffordfajardo,

I really appreciate any help and input so feel free to raise lots of bugs and questions.

Some tasks on the initial todo list (happy to recieve help on anything):

- [x] Fix jiti for the @remix-kit/react package.
- [x] Investigate if there's a way to avoid the need for RemixKitRunner and just run the server.(jt)s files through Vite Node.
- [x] Investigate supporting @remix-run/serve rather than just apps with custom server files.
- [ ] Performance profiling and optimization of the HMR.
- [x] Setup CI for build, lint, type checking and initial unit tests.
- [ ] Testing against real-world code bases of medium to large size.
- [ ] Changesets setup for this monorepo https://github.com/changesets/changesets
- [x] Automatically handle the CSS ?url suffix so changes aren't needed to Remix codebases (could be a good PR option). 
- [x] An npx package to automatically setup a remix repo for remix-kit with options like just development or build/development, vue/react etc.
- [x] Remove the need for the runner.ready() callback. Could be simpler to just configure a port/url to listen to. Probably depends on how item 2 goes.
- [ ] The devServer config isn't actually used. Either remove and prefer cmd line and env var overrides or get working.
- [ ] Preview command needs a rethink as isn't very useful at this stage.

Longer-term:

- [ ] Support default templates for entry.client.tsx, entry.server.tsx, root.tsx.
- [ ] Support Cloudflare's workerd by removing vite-node's dependency on node's vm.
- [ ] Module federation, async entry files.
- [ ] Vue SSR support (dependency on making a lot of @remix-run/react view agnostic).
- [ ] Support multiple Remix server-build objects and merge them at runtime. Allows splitting site into packages, module federation etc.

**Reply by cliffordfajardo** [reply]
> Module federation, async entry files.
>  Support multiple Remix server-build objects and merge them at runtime. Allows splitting site into packages, module federation etc.

How's the module federation support in the vite ecosystem?


Have you seen this ⬇️
- Remix Microfrontend implementation: https://github.com/remix-run/remix/discussions/1989
- ^Video: https://portal.gitnation.org/contents/scaling-up-with-remix-and-micro-frontends

**Reply by jrestall** [reply]
It's very good, there's [vite-plugin-federation](https://github.com/originjs/vite-plugin-federation) which is based on Webpack 5 Module Federation. However I'm more interested in [native federation](https://www.npmjs.com/package/@softarc/native-federation) since it's based on browser standard import maps and doesn't couple you long-term to a specific compiler. So my preference is https://github.com/module-federation/vite.

@adrienbaron has done awesome work in this space that I've kept an eye on. I have noticed that he's had to fork and patch the default remix compiler though, so I'm hoping having an extensible alternative will make supporting micro frontends a lot easier.
